extensions/script-kit/src/run-script.tsx

```diff
@@ -0,0 +1,97 @@
+import { resolve } from "path";
+import { useEffect, useState } from "react";
+import { Script } from "./script";
+
+const kit_path = resolve(homedir(), ".kit");
+
+export default function listScripts(): JSX.Element {
+  const [scripts, setScripts] = useState<Script[]>();
+  useEffect(() => {
+    const script_db_path = resolve(kit_path, "db", "scripts.json");
+    readFile(script_db_path)
+      .then((buffer) => buffer.toString())
+      .then(JSON.parse)
+      .then((json) => setScripts(json.scripts));
+  }, []);
+
+  return (
+    <List>
+      {scripts
+        ?.filter((script) => !script.exclude)
+        .map((script) => (
+          <List.Item
+            title={script.name}
+            key={script.command}
+            subtitle={script.description}
+            accessoryTitle={script.kenv}
+            actions={
+              <ActionPanel>
+                <ActionPanel.Section>
+                  <RunScriptAction scriptPath={script.filePath} />
+                </ActionPanel.Section>
+                <ActionPanel.Section>
+                  <NewScriptAction/>
+                  <NewScriptFromUrlAction/>
+                </ActionPanel.Section>
+                <ActionPanel.Section>
+                  <ShareScriptAsGistAction scriptPath={script.filePath}/>
+                  <EditScriptAction scriptPath={script.filePath}/>
+                  <DeleteScriptAction scriptPath={script.filePath}/>
+                </ActionPanel.Section>
+              </ActionPanel>
+            }
+          />
+        ))}
+    </List>
+  );
+}
+
+function DeleteScriptAction(props: { scriptPath: string }) {
+  return (
+    <RunScriptAction
+      title="Remove Script"
+      icon={Icon.Trash}
+      scriptPath={resolve(kit_path, "cli", "remove.js")}
+      args={[props.scriptPath]}
+    />
+  );
+}
+
+function NewScriptAction() {
+  return <RunScriptAction title="New Script" icon={Icon.Plus} scriptPath={resolve(kit_path, "cli", "new.js")} />;
+}
+
+function NewScriptFromUrlAction() {
+  return <RunScriptAction title="New Script from Url" icon={Icon.Link} scriptPath={resolve(kit_path, "cli", "new-from-url.js")} />;
+}
+
+function ShareScriptAsGistAction(props: { scriptPath: string }) {
+  return (
+    <RunScriptAction title="Share Script as Gist" icon={Icon.Upload} scriptPath={resolve(kit_path, "cli", "share-script.js")} args={[props.scriptPath]} />
+  );
+}
+
+function EditScriptAction(props: { scriptPath: string }) {
+  return (
+    <RunScriptAction title="Edit Script" icon={Icon.Pencil} scriptPath={resolve(kit_path, "cli", "edit.js")} args={[props.scriptPath]} />
+  );
+}
+
+function RunScriptAction(props: { scriptPath: string; icon?: Icon, title?: string; args?: string[] }) {
+  return (
+    <ActionPanel.Item
+      title={props.title || "Run Script"}
+      icon={props.icon || Icon.Terminal}
+      onAction={async () => {
+        const kar_path = resolve(kit_path, "kar");
+        await spawn(kar_path, [props.scriptPath, ...(props.args || [])]);
```

Comment:
`spawn` doesn't return a `Promise`, so `await` isn't doing anything.

https://nodejs.org/api/child_process.html#child_processspawncommand-args-options

You _could_ handle technically handle errors here, but since you're limiting to scripts found in the scripts.json, you would have to somehow delete a script while Raycast is open for this to error. I could even handle this on the Kit.app side, so I don't see any issues here.

Reply:
I switched to execa to be able to show a message if kit is not running :)